### DISCUSSION TITLE
Removing packages from original image results in build error

### DISCUSSION THREAD

**Discussion by arturasb** [blue-build/.github#36] [Support] [answered]
Hi.

I wanted my custom build of silverblue-main, mainly to add flatpak apps I need for my installations. I added instructions to remove couple of ostree packages, namely Firefox and LibreOffice in favor of flatpak-packed variants of the same apps:

```
remove:
      # example: removing firefox (in favor of the flatpak)
      # "firefox" is the main package, "firefox-langpacks" is a dependency
      - firefox
      - firefox-langpacks # also remove firefox dependency (not required for all packages, this is a special case)
      - libreoffice
```

Having this in my recipe.yml resulted in the following build error:
```
[16:57:37 g.i/a/my-silverblue:40] => ERROR: failed to solve: process "/bin/sh -c /tmp/scripts/run_module.sh 'rpm-ostree' '{\"type\":\"rpm-ostree\",\"repos\":null,\"install\":[\"vit-manager\"],\"remove\":[\"firefox\",\"firefox-langpacks\",\"libreoffice\"]}'   && ostree container commit" did not complete successfully: exit code: 1
[16:59:18 ERROR] => Failed:
  × Failed to build image ghcr.io/arturasb/my-silverblue:40

Error: Process completed with exit code 1.
```

I also checked the documentation on recipe.yml, file seemed to be OK. Could you point me to some information on how to resolve this ? Or maybe anybody here had similar issues and found solution ?

Regards
ArtūrasB.

**Comment by sukarn-m** [comment, marked as answer]
I'm not a developer of blue-build.
The actual error is a few lines above your quote from the log.
Taking [this action run](https://github.com/arturasb/mySilverblue/actions/runs/10434011909/job/28896386851) from your repository:
> error: No installed package matches 'libreoffice'

silverblue-main does not contain `libreoffice`, so rpm-ostree cannot uninstall it.

**Reply by arturasb** [reply]
Thank you  ! I managed not to notice error about missing libreoffice .
I wonder if there is a way to check what packages/applications are included to silverblue-main...

**Reply by sukarn-m** [reply]
On a running fedora-based system, like silverblue-main, you can run this command:
```bash
rpm -qa | sort | less
```

Alternatively, in the build steps, run this script as the first step:
```bash
rpm -qa | sort
exit 1
```
